tests/core/pyspec/eth2spec/test/electra/block_processing/test_process_withdrawals.py

```diff
@@ -94,14 +94,351 @@ def test_pending_withdrawals_one_skipped_one_effective(spec, state):
+        partial_withdrawals_indices=[],
+        pending_withdrawal_requests=[pending_withdrawal]
+    )
+
+    assert state.pending_partial_withdrawals == []
+
+
+@with_electra_and_later
+@spec_state_test
+def test_pending_withdrawals_with_ineffective_sweep_on_top_2(spec, state):
+    # Ensure validator will be processed by the sweep
+    validator_index = min(len(state.validators), spec.MAX_VALIDATORS_PER_WITHDRAWALS_SWEEP) // 2
+
+    pending_withdrawal_0 = prepare_pending_withdrawal(
+        spec, state,
+        validator_index,
+        effective_balance=spec.MAX_EFFECTIVE_BALANCE_ELECTRA,
+        amount=spec.EFFECTIVE_BALANCE_INCREMENT // 2
+    )
+
+    pending_withdrawal_1 = prepare_pending_withdrawal(
+        spec, state,
+        validator_index,
+        effective_balance=spec.MAX_EFFECTIVE_BALANCE_ELECTRA,
+        amount=spec.EFFECTIVE_BALANCE_INCREMENT
+    )
+
+    # Set excess balance in a way that validator
+    # becomes not partially withdrawable only after the second pending withdrawal is processed
+    state.balances[validator_index] = spec.MAX_EFFECTIVE_BALANCE_ELECTRA + spec.EFFECTIVE_BALANCE_INCREMENT
+    assert spec.is_partially_withdrawable_validator(
+        state.validators[validator_index],
+        state.balances[validator_index] - pending_withdrawal_0.amount
+    )
+    assert not spec.is_partially_withdrawable_validator(
+        state.validators[validator_index],
+        state.balances[validator_index] - pending_withdrawal_0.amount - pending_withdrawal_1.amount
+    )
+
+    next_slot(spec, state)
+    execution_payload = build_empty_execution_payload(spec, state)
+    yield from run_withdrawals_processing(
+        spec, state,
+        execution_payload,
+        num_expected_withdrawals=2,
+        fully_withdrawable_indices=[],
+        partial_withdrawals_indices=[],
+        pending_withdrawal_requests=[pending_withdrawal_0, pending_withdrawal_1]
+    )
+
+    assert state.pending_partial_withdrawals == []
+
+
+@with_electra_and_later
+@spec_state_test
+def test_pending_withdrawals_with_effective_sweep_on_top(spec, state):
+    # Ensure validator will be processed by the sweep
+    validator_index = min(len(state.validators), spec.MAX_VALIDATORS_PER_WITHDRAWALS_SWEEP) // 2
+
+    pending_withdrawal_0 = prepare_pending_withdrawal(
+        spec, state,
+        validator_index,
+        effective_balance=spec.MAX_EFFECTIVE_BALANCE_ELECTRA,
+        amount=spec.EFFECTIVE_BALANCE_INCREMENT // 2
+    )
+
+    pending_withdrawal_1 = prepare_pending_withdrawal(
+        spec, state,
+        validator_index,
+        effective_balance=spec.MAX_EFFECTIVE_BALANCE_ELECTRA,
+        amount=spec.EFFECTIVE_BALANCE_INCREMENT
+    )
+
+    # Set excess balance to requested amount times three,
+    # so the validator is partially withdrawable after pending withdrawal is processed
+    state.balances[validator_index] = spec.MAX_EFFECTIVE_BALANCE_ELECTRA + spec.EFFECTIVE_BALANCE_INCREMENT * 3
```

Comment:
I believe 2x is enough for this test to pass. I'd suggest using that instead.
```suggestion
    state.balances[validator_index] = spec.MAX_EFFECTIVE_BALANCE_ELECTRA + spec.EFFECTIVE_BALANCE_INCREMENT * 2
```